lib/web/ui/integration.go

```diff
@@ -168,10 +168,48 @@ func (r *Integration) CheckAndSetDefaults() error {
 	return nil
 }
 
+type IntegrationOAuthCredentials struct {
+	ID     string `json:"id"`
+	Secret string `json:"secret"`
+}
+
+func (c *IntegrationOAuthCredentials) Check() error {
+	if c.ID == "" {
+		return trace.BadParameter("missing OAuth ID field")
+	}
+	if c.Secret == "" {
+		return trace.BadParameter("missing OAuth Secret field")
+	}
+	return nil
+}
+
+type CreateIntegrationRequest struct {
+	Integration
+
+	OAuth *IntegrationOAuthCredentials `json:"oauth,omitempty"`
+}
+
+func (r *CreateIntegrationRequest) CheckAndSetDefaults() error {
+	if err := r.Integration.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+	if r.SubKind == types.IntegrationSubKindGitHub {
+		if r.OAuth == nil {
+			return trace.BadParameter("missing OAuth settings for GitHub integrations")
+		}
+		if err := r.OAuth.Check(); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
 // UpdateIntegrationRequest is a request to update an Integration
 type UpdateIntegrationRequest struct {
 	// AWSOIDC contains the fields for `aws-oidc` subkind integration.
-	AWSOIDC *IntegrationAWSOIDCSpec `json:"awsoidc,omitempty"`
+	AWSOIDC *IntegrationAWSOIDCSpec      `json:"awsoidc,omitempty"`
+	GitHub  *IntegrationGitHub           `json:"github,omitempty"`
```

Comment:
i'm not sure the github organization name should be made modifiable (i think we would rather want the user to just go through the integration flow for a different organization?)

also would changing the org name break the existing git server?

if we remove this field, we probably need to add a `integrationKind` field, to help communicate that the `oauth` is specifically for kind `github`?

Reply:
i can remove the org change part. At higher level (web flow) changing org may not make sense but it's technically doable, but we can keep that to `tctl`.

We retrieve `integrationKind` from the existing integration so there is no need for the update request.